dvc/remote/ssh/connection.py

```diff
@@ -185,8 +185,27 @@ def download(self, src, dest, no_progress_bar=False, progress_title=None):
             self.sftp.get(src, dest, callback=pbar.update_to)
 
     def move(self, src, dst):
+        """Rename src to dst, if it is not possible (in case src and dst are
+        on different filesystems) and actual physical copying of data is
+        happening.
+        """
         self.makedirs(posixpath.dirname(dst))
-        self.sftp.rename(src, dst)
+
+        try:
+            self.sftp.rename(src, dst)
```

Comment:
are you 100% sure that rename is never copying data? just to double-check

Reply:
If I understand the man page correctly, POSIX's rename is atomic: http://man7.org/linux/man-pages/man2/rename.2.html

But it turns out SFTP doesn't follow POSIX standard, there's an specific call `sftp.posix_rename` for that, it is implemented as an OpenSSH extension (widely used) so I guess it is fine to use it :)

Reply:
i was more asking about it failing if you try to rename from one mounted point to another - is is part of the standard as it is for regular posix rename

re atomicity - yeah, it looks like sftp rename is not atomic, but in this sense our own version with tmp file would not be better, so it's probably safe to use it 👍